### PR TITLE
Site Importer: Add a site preview step.

### PR DESCRIPTION
**Pull request by bisko**
This PR adds a better Site Preview step to the Site importer flow.

In the step we will show a screenshot of the site to be imported, so the user can be sure that this is what they want to import.

To test:

1. Check out branch or use Calypso.live link
2. Go to Site importer
3. Try to import a site
4. Verify that on the confirmation step you see a screenshot of the site you want to import
5. Verify no JS errors are generated and the import completes properly after that

### CODE REVIEW

**Comment by marekhrabe** [comment]
This PR works well and code looks good for me but I'm a bit concerned about the "supported content" removal. That way, users will automatically think that we will bring everything over but that's not the case - we have a set list of supported content. I think it was an important part of understanding what will happen.

**Comment by bisko** [comment]
> I'm a bit concerned about the "supported content" removal. That way, users will automatically think that we will bring everything over but that's not the case - we have a set list of supported content. I think it was an important part of understanding what will happen.

I agree here. The way I set it up is closely following the original design as it was created. I'd be happy to update it to include what will be imported, but I'm not sure what will be the best way to approach that and not create too much visual noise.

**Comment by ianstewart** [comment]
Current off-the-cuff thinking: Might be best towards the end of the process. A, "here's what we imported for you" type checklist message.

**Comment by marekhrabe** [comment]
To me, it seems like too late to show the list only after the import finishes. Users can leave the import process (we even encourage that in the copy I think) and they might never come back to the import screen. However, the more important part for me is the expectation - finding out we didn't import their shop after they went through the whole process might be a frustrating experience. 

I think we should have a list of supported content before you start and some kind of end-screen (TBD) with the actual list of things that got imported. 

I was thinking about bringing the list back here and showing it next to the screenshot in similar way:
screen shot 2018-05-15 at 15 16 13

Do you think this could work?

File: client/my-sites/importer/site-importer/site-importer-input-pane.jsx [hunk 1]
```diff
@@ -112,6 +114,11 @@ class SiteImporterInputPane extends React.Component {
 
 		this.setState( { loading: true }, this.resetErrors );
 
+		request
```
[the remark this answers is not in the document]
Comment (bisko) [reply client/my-sites/importer/site-importer/site-importer-input-pane.jsx]: Thanks for the nudge. I extracted the call to an external utility "library". We can probably move it to a more general place if we see the need to reuse it somewhere ( we probably could update pieces of code here and there ).

File: client/my-sites/importer/site-importer/site-importer-input-pane.jsx [hunk 2]
```diff
@@ -112,6 +114,11 @@ class SiteImporterInputPane extends React.Component {
...
+			.get( `https://s0.wp.com/mshots/v1/${ siteURL }` )
+			.then( noop )
```
[the remark this answers is not in the document]
Comment (bisko) [reply client/my-sites/importer/site-importer/site-importer-input-pane.jsx]: I wanted to be a bit more explicit :) In the updated version the `then` `noop` is removed and only the `catch` `noop` is left as we don't care about any errors that may arise in that context and it's used just to start a preloading process.

File: client/my-sites/importer/style.scss [hunk 3]
```diff
@@ -219,16 +219,19 @@
...
 	}
 
-
-
 	& .site-importer__site-importer-confirm-site-pane-container {
 		display: flex;
-		flex-direction: row;
 		align-content: center;
-		justify-content: flex-end;
+		justify-content: center;
```
Comment (jblz) [inline client/my-sites/importer/style.scss]: Can we hide the `Is this your site?` prompt & buttons (`.site-importer__site-importer-confirm-site-pane-container`) until the image is fully loaded (or the Promise rejects)?

Not sure if the wifi is slow where I am or if it took a bit to process, but I saw a screen that looked like this for 5ish seconds the first few times I tried this:

screen shot 2018-05-24 at 12 54 06 pm

Comment (bisko) [reply client/my-sites/importer/style.scss]: > Can we hide the Is this your site? prompt & buttons (.site-importer__site-importer-confirm-site-pane-container) until the image is fully loaded (or the Promise rejects)?

I like this. Updated the code to reflect that.

File: client/my-sites/importer/site-importer/site-preview-actions.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+					currentRetries: currentRetries + 1
+				} ), retryTimeout );
+			}
+			else if ( res.type === 'image/jpeg' ) {
+				// Successfully generated the preview
+				try {
+					resolve( window.URL.createObjectURL( res.xhr.response ) );
+				}
+				catch ( e ) {
+					reject( e );
```
[the remark this answers is not in the document]
Comment (bisko) [reply client/my-sites/importer/site-importer/site-preview-actions.js]: Thanks for the tip! :) Updated!

**Pull request merged**